src/meta/src/controller/streaming_job.rs

```diff
@@ -541,12 +541,32 @@ impl CatalogController {
             return Err(MetaError::permission_denied("table version is stale"));
         }
 
+        // 2. check concurrent replace.
+        let replacing_cnt = ObjectDependency::find()
+            .join(
+                JoinType::InnerJoin,
+                object_dependency::Relation::Object1.def(),
+            )
+            .join(JoinType::InnerJoin, object::Relation::StreamingJob.def())
+            .filter(
+                object_dependency::Column::Oid
+                    .eq(id as ObjectId)
+                    .and(object::Column::ObjType.eq(ObjectType::Table))
+                    .and(streaming_job::Column::JobStatus.ne(JobStatus::Created)),
+            )
+            .count(&txn)
+            .await?;
+        if replacing_cnt != 0 {
```

Comment:
Is it possible that there'a race case where multiple threads pass this check, as we don't do this in the same transaction as insertion of job obj?

Reply:
No, the function is protected by a read-write lock inside the catalog controller.